app/controllers/rooms_controller.rb

```diff
@@ -1,115 +1,167 @@
+    params.require(:room).permit(:room_id, :description, selected_issues: [])
   end
 
   def init_items_size_list
-    @items_size_list ||= [Room.all.size, Room.where("is_clean = ?", false).size, Room.where("is_clean = ?", true).size]
-    #                             0                           1                                       2
-  end 
+    @items_size_list ||= [Room.all.size, Room.where('is_clean = ?', false).size,
+                          Room.where('is_clean = ?', true).size]
+  end
 
   def report_modal
-    room = params[:room_id]
-    @room_id = room
-    @issuesForRoom = Issue.where("room_id = ?", room)
-
-    @issues = Array.new
-
-    @issuesForRoom.each do |issue| 
-      issueEntity = IssueEntity.new
-      type = IssueType.where("id = ?", issue.issue_type_id).first
-      issueEntity.issue_type = issue.issue_type_id
-      issueEntity.room_id =@room
-      issueEntity.icon_path = type.icon_path
-      issueEntity.is_done = issue.is_done
-      issueEntity.issue_description = type.issue_description
-
-      @issues.push(issueEntity)
+    @room_id = params[:room_id]
+    @issues = create_issue_entities(@issues_for_room)
+  end
+
+  def create_issue_entities(issues_for_room)
+    issues = []
+    issues_for_room.each do |issue|
+      type = IssueType.where('id = ?', issue.issue_type_id).first
+      issue_entity = IssueEntity.new(issue.issue_type_id, params[:room_id],
+                                     issue.is_done, type.issue_description,
+                                     type.icon_path)
+      issues.push(issue_entity)
     end
+    return issues
   end
 
   def report_issue
     room_id = params[:room][:room_id]
-    issueType = params[:room][:issue_type]
-    @issue = Issue.where("room_id = ? AND issue_type_id = ?", room_id, issueType).first
-    @room = Room.where("room_id = ?", room_id).first
-    currentValue = @issue.is_done
-    @issue.update_attribute(:is_done, !currentValue)
-    @issue.update_attribute(:fix_comment, params[:room][:comment])
-    @issue.update_attribute(:completion_date, Date.today)
-    @room.update_attribute(:is_clean, false)
-
+    issue_type = params[:room][:issue_type]
+    issue = Issue.where('room_id = ? AND issue_type_id = ?',
+                        room_id, issue_type).first
+    room = Room.where('room_id = ?', room_id).first
+    current_value = issue.is_done
+    issue.update_attributes(is_done: !current_value,
+                            fix_comment: params[:room][:comment],
+                            completion_date: Date.today)
+    room.update_attribute(:is_clean, false)
     redirect_back(fallback_location: root_path)
   end
+
+  # Helper methods
+
+  def fetch_issues_by_room_id
+    @issues_for_room = Issue.where('room_id = ?', params[:room_id])
+  end
+
+  def fetch_all_rooms
+    @rooms = Room.all
+  end
+
+  def fetch_selected_issues
+    @selected_issues = Issue.where(room_id: params[:id])
```

Comment:
Yes, we can - I'll put it in another refactor. It is, however, out of scope for this one as I also need to adjust the parameters passed from views.